manager/integration/tests/test_backing_image.py

```diff
@@ -694,3 +704,199 @@ def test_volume_wait_for_backing_image_condition(client): # NOQA
+                         backingImage=BACKING_IMAGE_NAME)
+    vol = wait_for_volume_detached(client, volume_name)
+    assert vol.diskSelector == volume_disk_selector
+    assert vol.nodeSelector == volume_node_selector
+
+    assert vol.conditions.Scheduled.status == "False"
+    cleanup_all_volumes(client)
+    cleanup_all_backing_images(client)
+
+
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_eviction(client): # NOQA
+    """
+    Test Node Eviction
+    - Create BackingImage with one copy
+    - Evict the node where the copy is on
+    - The BackingImage will be evicted to another node
+
+    Test Disk Eviction
+    - Evict the disk where the copy is currently on
+    - The BackingImage will be evicted to other node
+      because every node has only one disk
+    """
+
+    bi_url = BACKING_IMAGE_RAW_URL
+    create_backing_image_with_matching_url(
+        client, BACKING_IMAGE_NAME, bi_url,
+        minNumberOfCopies=1)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    disk_id_1 = next(iter(backing_image.diskFileStatusMap))
+
+    node_1 = get_node_by_disk_id(client, disk_id_1)
+    node_1 = set_node_scheduling_eviction(
+        client, node_1, allowScheduling=False, evictionRequested=True)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+    check_backing_image_single_copy_node_eviction(
+        client, BACKING_IMAGE_NAME, node_1)
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    disk_id_2 = next(iter(backing_image.diskFileStatusMap))
+    node_2 = get_node_by_disk_id(client, disk_id_2)
+
+    # Set node1 back to allow scheduling
+    node_1 = set_node_scheduling_eviction(
+        client, node_1, allowScheduling=True, evictionRequested=False)
+
+    # Evict the disk on the node2
+    # and check the backing image file is evicted to node1
+    update_disks = get_update_disks(node_2.disks)
+    for disk in update_disks.values():
+        disk.allowScheduling = False
+        disk.evictionRequested = True
+
+    node_2 = update_node_disks(client, node_2.name, disks=update_disks,
+                               retry=True)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+    check_backing_image_single_copy_disk_eviction(
+        client, BACKING_IMAGE_NAME, disk_id_2)
+
+    cleanup_all_backing_images(client)
+
+
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_unable_eviction(client): # NOQA
+    """
+    - Set node1 nodeTag = [node1], diskTag = [node1]
+    - Create a BackingImage with following settings to place the copy on node1
+        - minNumberOfCopies = 1
+        - nodeSelector = [node1]
+        - diskSelector = [node1]
+    - Evict node1
+    - The copy would not be deleted because it is the only copy
+    - The copy can't be copied to other nodes because of the selector settings.
```

Comment:
What can a user do for the case? Will it introduce a different behavior than before?

Reply:
no, we don't have backing image eviction feature before
so when user evict a node, it won't even try to create the copy to other nodes in the previous version

This behavior is to prevent users from draining the node without noticing that this is the last backing image file
There is a event for this, and users should notice that the backing image is not evicted.

Moreover, the behavior is exactly the same as replicas
https://github.com/longhorn/longhorn-manager/blob/dcd5164185d7130732f72bb2e978dfdc0338acd1/controller/volume_controller.go#L1127-L1146